odoo/addons/base/models/ir_ui_view.py

```diff
@@ -1447,3 +1447,150 @@ def _validate_module_views(self, module):
                 view._check_xml()
             except Exception as e:
                 self.raise_view_error("Can't validate view:\n%s" % e, view.id)
+
+    def _fix_faulty_views(self, delete=False):
+        """ Use the _get_faulty_views and take appropriate actions to fix the issue """
+        for view in self:
+            view_id = view.id # allow to use even if view is deleted
+            faulty_views = view._get_faulty_views(view,self.env['ir.ui.view'])
+            if faulty_views:
+                self._cr.execute('SAVEPOINT end_validation')
+                faulty_views._clean_properly(delete)
+                self._cr.execute("select count(id) from ir_ui_view where id = %s", (view_id,))
+                cnt = self.env.cr.fetchone()[0]
+                if cnt == 0 or view._is_valid():
+                    _logger.info("View %s can be fixed by disabling/deleting following views: %s" % (view_id,faulty_views))
+                    self._cr.execute('RELEASE end_validation')
+                else:
+                    self._cr.execute('ROLLBACK TO end_validation')
+                    self._cr.execute('RELEASE end_validation')
+                    _logger.warning("Issue occurs: The proposition made to fix the view %s (disable/delete %s) is not recommended" % (view,faulty_views))
+                    # should not happend
+
+    def _get_faulty_views(self,parent_view, disabled_views):
```

Comment:
You could do
```python
def _get_faulty_views(self,parent_view=None):
if parent_view is None:
    parent_view = self
    self._cr.execute('SAVEPOINT test_fix_views')
```
weird to pass empty recordset as arg

Reply:
to check:
avoid savepoint and only work in cache with `update({})`?

Reply:
Not really motivated to do it:
I don't really know from where this will be called. (from saas-migration but not only) if called from a custom server action, I prefer be sure that we rollback to savepoint instead of using updating only the cache and the issue that comes with it.

Reply:
done